### PR TITLE
Make Podfiles use symlinks to local pods

### LINKED ISSUE #9502: A Flutter app's Podfile.lock should be versionable
## Steps to Reproduce

Create a new app and build it for the simulator:

```
flutter create myapp
cd myapp
<add plugin dependency to pubspec.yaml>
flutter run -d i
git init
git add .
git commit
```

You've just committed a `Podfile.lock` file that contains local paths to your plugins and Flutter framework, e.g.

```
EXTERNAL SOURCES:
  Flutter:
    :path: /Users/jackson/git/flutter/bin/cache/artifacts/engine/ios
```

Everyone who builds the app will end up with a different `Podfile.lock`, and won't be sure whether to check in their version. Unfortunately, [adding `Podfile.lock` to `.gitignore` is not recommended by the CocoaPods team](http://guides.cocoapods.org/using/using-cocoapods.html#benefits-of-ignoring-the-pods-directory) because it makes builds non-reproducible.

I think we can address this by making some changes to the template app's [`Podfile`](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/templates/create/ios.tmpl/Podfile).

Instead of having `.flutter-plugins` be a file, have it be a directory that contains contains a symbolic link called Flutter pointing to the Flutter framework directory, as well as symbolic links to the directories of all the plugin Pods.

Now the Flutter section of the Podfile looks like this:

```
  # Flutter Pods
  if Dir.exists? '../.flutter-plugins'
    Dir.foreach('../.flutter-plugins') { |name|
      next if name == '.' or name == '..'
      pod name, :path => "../.flutter-plugins/#{name}/ios"
    }
  end
```

### Benefits

  * Less boilerplate code in the Podfile
  * Going to the `ios/` directory and running `pod install` manually will actually work (after the first Flutter iOS build) instead of complaining that the `FLUTTER_FRAMEWORK_DIR` environment variable is not set.
  * The `Podfile.lock` will be the same for everyone. It will look like this:

```
DEPENDENCIES:
  - firebase_database (from `../.flutter-plugins/firebase_database/ios`)
  - firebase_storage (from `../.flutter-plugins/firebase_storage/ios`)
  - Flutter (from `../.flutter-plugins/Flutter/ios`)
  - google_sign_in (from `../.flutter-plugins/google_sign_in/ios`)
  - image_picker (from `../.flutter-plugins/image_picker/ios`)

EXTERNAL SOURCES:
  firebase_database:
    :path: ../.flutter-plugins/firebase_database/ios
  firebase_storage:
    :path: ../.flutter-plugins/firebase_storage/ios
  Flutter:
    :path: ../.flutter-plugins/Flutter/ios
  google_sign_in:
    :path: ../.flutter-plugins/google_sign_in/ios
  image_picker:
    :path: ../.flutter-plugins/image_picker/ios
...
```

While we are at it, I think we should add `Pods/` to `ios/.gitignore`, which is allowed according to the Cocoapods team. But that's more a matter of taste since (at least in theory) the contents of that directory should be the same for every developer as long as their Flutter engines are in sync.

### PR DESCRIPTION
**Pull request by mravn-google**
Fixes https://github.com/flutter/flutter/issues/9502

### CODE REVIEW

**Review by jakobr-google — APPROVED** [review]
LGTM

**Comment by xster** [comment]
Awesome, cool!
LGTM 

I wonder if we can put these symlinks in the /Pods directory to keep them bundled. Wonder if `pod deintegrate` would clean up everything nicely for us if it were inside /Pods.

**Comment by mravn-google** [comment]
> I wonder if we can put these symlinks in the /Pods directory to keep them bundled. This way, it 'self documents' too when the users are exploring the directories. Wonder if pod deintegrate would clean up everything nicely for us if it were inside /Pods.

Yes, I'll give it a try at least. One may worry that we'll be making assumptions about the detailed workings of `pod install`. Changes to the timing or the extend to which it clears the `Pods/` folder could affect correctness down the road.

**Comment by mravn-google** [comment]
* Moved `Symlinks` folder to `Pods/.symlinks`
* Now copying Flutter framework's LICENSE file to sibling of Flutter framework podspecs in cache. This avoids issue with resolving `../../../../../` relative to symlink.

Awaits https://github.com/flutter/engine/pull/4691 which updates the LICENSE placement in Flutter.podspec.

**Comment by mravn-google** [comment]
@xster PTAL

File: packages/flutter_tools/templates/create/ios.tmpl/.gitignore [hunk 1]
```diff
@@ -40,3 +40,4 @@ Icon?
 /ServiceDefinitions.json
 
 Pods/
+Symlinks/
```
Comment (jakobr-google) [inline packages/flutter_tools/templates/create/ios.tmpl/.gitignore]: Don't need this anymore, right?

**Review by jakobr-google — APPROVED** [review]
LGTM

File: packages/flutter_tools/templates/create/ios.tmpl/.gitignore (hunk 1 above)
Comment (mravn-google) [reply packages/flutter_tools/templates/create/ios.tmpl/.gitignore]: Good catch, thanks!

**Pull request merged**